src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,247 @@
+// eslint-disable-next-line
+import { useParams } from 'react-router';
+import { useEffect, useState } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
 
-export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+export const PeoplePage: React.FC<{
+  isError: boolean;
+  setIsError: React.Dispatch<React.SetStateAction<boolean>>;
+  isLoading: boolean;
+  peopleFromServer: Person[] | undefined;
+  errorMessage: string | undefined;
+}> = ({
+  isError, setIsError, isLoading, peopleFromServer, errorMessage,
+}) => {
+  const { slug = '' } = useParams();
+  const [visiblePeople, setVisiblePeople] = useState(peopleFromServer || []);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const initialSort = searchParams.get('sort');
+  const [sortUpdate, SetSortUpdate] = useState(initialSort);
+  const [sortBefore, SetSortBefore] = useState(sortUpdate);
+  const initialOrder = searchParams.get('order') || null;
+  const [orderUpdate, SetOrderUpdate] = useState(initialOrder);
+  const sortLane = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+  const setSearchWith = (
```

Comment:
inside this function, you don't use any value from closure - :+1: 
It allows you to move it outside of the component (or even in a separate file - helpers.ts, etc.)

Reply:
![image](https://github.com/mate-academy/react_people-table-advanced/assets/101456317/6e21109c-8662-44ea-8487-59704c1ec4be)